src/executor/jit.rs

```diff
@@ -41,19 +41,22 @@ impl std::fmt::Debug for JitNativeExecutor<'_> {
 }
 
 impl<'m> JitNativeExecutor<'m> {
-    pub fn from_native_module(native_module: NativeModule<'m>, opt_level: OptLevel) -> Self {
+    pub fn from_native_module(
+        native_module: NativeModule<'m>,
+        opt_level: OptLevel,
+    ) -> Result<Self> {
         let NativeModule {
             module,
             registry,
             metadata,
         } = native_module;
 
-        Self {
+        Ok(Self {
             engine: create_engine(&module, &metadata, opt_level),
             module,
             registry,
             gas_metadata: metadata.get::<GasMetadata>().cloned().unwrap(),
-        }
```

Comment:
Fixing this unwrap would require to merge the PR that removes unwraps from the metadata first